src/nvim/screen.c

```diff
@@ -5847,30 +5840,16 @@ next_search_hl_pos(
   return 0;
 }
 
-static void screen_start_highlight(int attr)
-{
-  screen_attr = attr;
-  ui_start_highlight(attr);
-}
-
-static void screen_stop_highlight(void)
-{
-  ui_stop_highlight();
```

Comment:
I would vote to keep at least the `ui_` "start"/"stop" pair (`ui_start_highlight`, `ui_stop_highlight`). It gives a semantic hint, and leaves clues when comparing Vim code, at a negligible cost.

Reply:
But the semantics are not "start" and "stop", the nvim UI layer can directly _set_ the highlight from whatever value to whatever other value. The old names made sense in vim where screen.c directly wrote to the terminal, and clear highlight probably was a prerequisite for some term.c operations. But in nvim tui.c does its own housekeeping when explictly `^[...m` needs to be sent or not, so it makes no sense: highlights only needs to be _set_ before explicitly grid modifying operations. 

We can have `ui_clear_highlight() { ui_set_highlight(0) }` (which rhymes nicely with that they are used for clear operations) and eliminate the ad-hoc constant, but we shouldn't be afraid to rename functions when semantics in fact are different from vim.

Reply:
Definitely. If the semantics changed then that was just my misunderstanding.